webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTable.js

```diff
@@ -78,40 +77,34 @@ class SubscriptionsTable extends Component {
     const allSubscriptionResults = this.props.subscriptions.results;
 
     const checkAllRowsSelected = () =>
-      allSubscriptionResults.length === this.state.selectedRows.length;
+      allSubscriptionResults.length === this.props.selectedRows.length;
 
-    const updateDeleteButton = () => {
-      this.props.toggleDeleteButton(this.state.selectedRows.length > 0);
-    };
     return ({
       allRowsSelected: () => checkAllRowsSelected(),
       selectAllRows: () => {
         if (checkAllRowsSelected()) {
-          this.setState(
-            { selectedRows: [] },
-            updateDeleteButton,
-          );
+          this.props.onSelectedRowsChange([]);
+          this.props.toggleDeleteButton(false);
         } else {
-          this.setState(
-            { selectedRows: allSubscriptionResults.map(row => row.id) },
-            updateDeleteButton,
-          );
+          this.props.onSelectedRowsChange(allSubscriptionResults.map(row => row.id));
+          this.props.toggleDeleteButton(true);
         }
       },
       selectRow: ({ rowData }) => {
-        let { selectedRows } = this.state;
+        let { selectedRows } = this.props;
         if (selectedRows.includes(rowData.id)) {
           selectedRows = selectedRows.filter(e => e !== rowData.id);
         } else {
           selectedRows.push(rowData.id);
```

Comment:
The bug we talked about on IRC:

```
Uncaught ImmutableError {name: "MyError", message: "The push method cannot be invoked on an Immutable data structure.", stack: "Error↵    at new ImmutableError (https://centos7-k…0.8-development-2be37b10db4fa73cba5e.js:236922:5)"}
```

We can fix it like this, I think. Try it out by replacing the .push with:

```
selectedRows = selectedRows.concat(rowData.id);
```